torchvision/prototype/transforms/_geometry.py

```diff
@@ -2,107 +2,104 @@
 import math
 import numbers
 import warnings
-from typing import Any, Dict, List, Union, Sequence, Tuple, cast
+from typing import Any, Dict, List, Optional, Union, Sequence, Tuple, cast
 
 import PIL.Image
 import torch
 from torchvision.prototype import features
 from torchvision.prototype.transforms import Transform, functional as F
 from torchvision.transforms.functional import pil_to_tensor, InterpolationMode
-from torchvision.transforms.transforms import _setup_size, _interpolation_modes_from_int
+
+# TODO: refactor _parse_pad_padding into
+# torchvision.transforms.functional and update F_t.pad and F_pil.pad
+# and remove redundancy
+from torchvision.transforms.functional_tensor import _parse_pad_padding
+from torchvision.transforms.transforms import _setup_size, _setup_angle, _check_sequence_input
 from typing_extensions import Literal
 
 from ._transform import _RandomApplyTransform
 from ._utils import query_image, get_image_dimensions, has_any, is_simple_tensor
 
 
 class RandomHorizontalFlip(_RandomApplyTransform):
-    def _transform(self, input: Any, params: Dict[str, Any]) -> Any:
-        if isinstance(input, features.Image):
-            output = F.horizontal_flip_image_tensor(input)
-            return features.Image.new_like(input, output)
-        elif isinstance(input, features.SegmentationMask):
-            output = F.horizontal_flip_segmentation_mask(input)
-            return features.SegmentationMask.new_like(input, output)
-        elif isinstance(input, features.BoundingBox):
-            output = F.horizontal_flip_bounding_box(input, format=input.format, image_size=input.image_size)
-            return features.BoundingBox.new_like(input, output)
-        elif isinstance(input, PIL.Image.Image):
-            return F.horizontal_flip_image_pil(input)
-        elif is_simple_tensor(input):
-            return F.horizontal_flip_image_tensor(input)
+    def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
+        if isinstance(inpt, features._Feature):
+            return inpt.horizontal_flip()
+        elif isinstance(inpt, PIL.Image.Image):
+            return F.horizontal_flip_image_pil(inpt)
+        elif is_simple_tensor(inpt):
```

Comment:
Nit: we could avoid the `is_simple_tensor` and just check for isinstance(..., Tensor) now.